api/proto/teleport/healthcheckconfig/v1/health_check_config.proto

```diff
@@ -0,0 +1,63 @@
+// Copyright 2025 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.healthcheckconfig.v1;
+
+import "google/protobuf/duration.proto";
+import "teleport/header/v1/metadata.proto";
+import "teleport/label/v1/label.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/healthcheckconfig/v1;healthcheckconfigv1";
+
+// HealthCheckConfig is the configuration for network health checks from an
+// agent to its proxied resource.
+message HealthCheckConfig {
+  string kind = 1;
+  string sub_kind = 2;
+  string version = 3;
+  teleport.header.v1.Metadata metadata = 4;
+
+  HealthCheckConfigSpec spec = 5;
+}
+
+// HealthCheckConfigSpec is the health check spec.
+message HealthCheckConfigSpec {
+  // Match is used to select resources that these settings apply to.
+  Matcher match = 1;
+  // Timeout is the health check connection establishment timeout.
+  // An attempt that times out is a failed attempt.
+  google.protobuf.Duration timeout = 2;
+  // Interval is the time between each health check.
+  google.protobuf.Duration interval = 3;
+  // HealthyThreshold is the number of consecutive passing health checks after
+  // which a target's health status becomes "healthy".
+  uint32 healthy_threshold = 4;
+  // UnhealthyThreshold is the number of consecutive failing health checks after
+  // which a target's health status becomes "unhealthy".
+  uint32 unhealthy_threshold = 5;
+}
+
+// Matcher is a resource matcher for health check config.
+message Matcher {
+  // DBLabels selects the databases to match.
+  // The result is logically ANDed with DBLabelsExpression.
+  // An empty value matches all databases.
```

Comment:
> // An empty value matches all databases.

Are we populating wildcard when `db_labels` is empty? or do you mean when both db_labels and db_labels_expression are empty? rfd says `The default for an empty matcher is that it matches nothing.`